src/clients/inst_add-on.rb

```diff
@@ -161,6 +162,46 @@ def current_addons
         Pkg.SourceURL(addon["media"])
       end
     end
+
+    # URL schemas to be considered as CD/DVD
+    DVD_SCHEMES = ["cd", "dvd"].freeze
+
+    # Determine whether the URL points to the installation CD/DVD
+    #
+    # @param url [String] Add-on URL
+    # @return [Boolean] true if it points to the installation CD/DVD
+    def instsys_dvd?(url)
+      scheme = URL.Parse(url)["scheme"]
+      return false unless DVD_SCHEMES.include?(scheme)
+
+      addon_device = device_from_url(url)
+      instsys_device && addon_device && addon_device == instsys_device
+    end
+
+    # instsys device
+    #
+    # @return [String] instsys device; nil if the device could not be
+    #   determined (for instance, when using a network based installation media)
+    def instsys_device
+      @instsys_device ||= device_from_url(InstURL.installInf2Url)
+    end
+
+    # Real device from an URL
+    #
+    # @param url [String] URL
+    # @return [String,nil] Path to the device; nil if no device is specified
+    #   or the device does not exist.
+    def device_from_url(url)
+      parsed = URL.Parse(url)
+      params = URL.MakeMapFromParams(parsed["query"])
+      return nil unless params["devices"]
+
+      begin
+        File.realpath(params["devices"])
```

Comment:
This won't work, the `devices` parameter, as the name suggests, is actually a list!

I.e. a valid URL is `dvd:///?devices=/dev/sr0,/dev/sr1`. Libzypp in this case tries the devices one by one and uses the first one which succeeds.